# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmCodegenTypes.cpp
# Commit: a0e51f584680
# Full Hash: a0e51f584680a017fc4d8124c58a51f3caadfade
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2021-09-16 09:43:03
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Introduce indirect stubs to optimize call_indirect. r=lth
#   
#   This patch introduces indirect stubs.
#   An indirect stub is a stub that takes care of any switching activities needed for call_indirect.
#   Before this patch, we have to conservatively assume that any call_indirect's target can be from a foreign instance,
# ==============================================================================

diff -r ece3dcc343a2 -r a0e51f584680 js/src/wasm/WasmCodegenTypes.cpp
--- a/js/src/wasm/WasmCodegenTypes.cpp	Thu Sep 16 04:38:26 2021 +0000
+++ b/js/src/wasm/WasmCodegenTypes.cpp	Thu Sep 16 05:09:24 2021 +0000
@@ -139,7 +139,7 @@
   u.func.lineOrBytecode_ = 0;
   u.func.beginToUncheckedCallEntry_ = 0;
   u.func.beginToTierEntry_ = 0;
-  MOZ_ASSERT(isEntry());
+  MOZ_ASSERT(isEntry() || isIndirectStub());
   MOZ_ASSERT(begin_ <= end_);
 }
 